projects/observability/src/shared/components/cartesian/chart.ts

```diff
@@ -63,9 +63,13 @@ export interface Band<TInterval> {
   hide?: boolean;
 }
 
-export interface GroupByFilter {
-  attribute?: FilterAttribute;
-  metadata: FilterAttribute[];
+export interface GroupBy {
```

Comment:
I would suggest not defining these two in chart.ts, since the concepts are not limited (or even used directly by) charts. Further, do we need new types or could we just update FilterAttribute to have an optional subpath for example (not sure on the scope of that change). Another relevant type would be `AttributeExpression` which models an attribute key and optional subpath. Basically I'm just trying to avoid us introducing a new type for each new usage when the data is the same.

Reply:
Agreed, I prefer to use AttributeExpression. Do you agree? I need to extract the reference but straightforward.

Reply:
I agree that's all that's needed, so if that can be used without a big refactor that seems to make sense. If you needed to wrap it (say, updating FilterAttribute's reference string to be an AttributeExpression) that could work too.

Reply:
done